lametro/events.py

```diff
@@ -492,92 +496,87 @@ def find_approved_minutes(self, event):
         if name not in {"Board of Directors - Regular Board Meeting", "LA SAFE"}:
             return None
 
-        # if the event is the future, there won't have been a chance to
+        # if the event is in the future, there won't have been a chance to
         # approve the minutes
         if event["start"] > datetime.datetime.now(datetime.timezone.utc):
             return None
 
         date = event["start"].strftime("%B %-d, %Y")
+
+        associated_with_meeting_body = f"MatterBodyId eq {event['EventBodyId']}"
+        meeting_date_in_title = f"substringof('{date}', MatterTitle)"
+        matter_type_minutes = "MatterTypeName eq 'Minutes'"
+        minutes_in_title = "substringof('Minutes', MatterTitle)"
+        matter_type_informational = "MatterTypeName eq 'Informational Report'"
+
         result = self.search(
             "/matters/",
             "MatterId",
-            "MatterBodyId eq {} and substringof('{}', MatterTitle) and substringof('Minutes', MatterTitle)".format(
-                event["EventBodyId"], date
+            (
+                f"{associated_with_meeting_body} and " +
+                f"{meeting_date_in_title} and " +
+                "(" +
+                    f"({matter_type_minutes}) or " +
+                    f"({minutes_in_title} and {matter_type_informational})" +
+                ")"
             ),
         )
 
-        try:
-            (matter,) = result
-        except ValueError as e:
-            if "not enough values" in str(e):
-                self.warning(
-                    "Couldn't find minutes for the {} meeting of {}.".format(name, date)
-                )
-                return None
-            elif "too many values to unpack" in str(e):
-                self.warning(
-                    "Found more than one minutes file for the {} meeting of {}.".format(
-                        name, date
-                    )
-                )
-                return None
-            else:
-                raise
+        # Will print a warning if no minutes have been found
+        n_minutes = 0
 
-        attachment_url = self.BASE_URL + "/matters/{}/attachments".format(
-            matter["MatterId"]
-        )
+        # Sometimes, the search returns more than one board report.
+        # Go through each matter yielded from this generator to account for that.
+        for matter in result:
+            if (matter['MatterRestrictViewViaWeb'] or
+            matter['MatterStatusName'] == 'Draft' or
+            matter['MatterBodyName'] == 'TO BE REMOVED'):
```

Comment:
Is this the same logic that's used by the bill scraper? If so, what about keeping it on the bill scraper, but making it a `@classmethod`, so it can be used here, as well? This logic has changed in the past, and I don't want to run the risk of forgetting to update it here.

Reply:
It's close but not exact. There's a couple extra conditions in the bill scraper, one of which is `or not matter.get('legistar_url')`.

Is this present on events? I've tested a handful of them and haven't seen any have this `legistar_url` attribute, which would incorrectly mark them as restricted.

I could split off the part I have in there now into a util file so it can be used in bills and events, maybe?

Reply:
Ah, if it's not the same, then let's keep it separate.